src/utils/env.ts

```diff
@@ -21,18 +22,25 @@ const PC_ENV_TLDS = environments
         return tlds;
     }, [] as string[])
     .concat(__PC_DEV_ENVS__);
-const GC_ENV_NAMES = new Set<string>([...environments, ...__GC_DEV_EXTRA_ENVS__].map((e) => e.name));
 
 const [defaultEnv] = environments.filter(env => env.region === DEFAULT_ENV_REGION);
 
 export const DEFAULT_PC_ENV = buildPcEnv(defaultEnv.publicDomainName);
 
-function isKnownEnvName(toCheck: string) {
-    return GC_ENV_NAMES.has(toCheck);
+function isKnownEnvName(toCheck: string, envs: Environment[]) {
+    const envList = new Set<string>([...envs, ...__GC_DEV_EXTRA_ENVS__].map((e) => e.name));
+    return envList.has(toCheck);
 }
 
-function findPcEnvironment(location: URL, targetEnv: string, parseEnvironment: EnvironmentParser): PcEnv|null {
-    const parsedEnv = parseEnvironment(location.origin, {});
+const matchesHostname = (hostname: string) => (domain: string) => {
+    return hostname === domain || hostname.endsWith(`.${domain}`);
+};
+
+function findPcEnvironment(location: URL, targetEnv: string, envs: Environment[]): PcEnv|null {
+    const parsedEnv = envs.find(({ publicDomainName, publicDomainAliases }) => {
```

Comment:
nit: since the logic is the same for both search blocks, should we setup the full env list first and then find?  Should be logically the same since we want to search for the default envs first.

```
[envs, ...__GC_DEV_EXTRA_ENVS__].find(...)
```